### PR TITLE
releng: Teach jobs using latest-fast marker to extract from K8s Infra

### PR DESCRIPTION
**Pull request by justaugustus**
As we continue to migrate release-blocking jobs to a dedicated K8s Infra
cluster, jobs that use the latest-fast marker need to extract builds
from `gs://k8s-release-dev`, which is the K8s Infra equivalent of
`gs://kubernetes-release-dev`.

A new flag (`--extract-ci-bucket=k8s-release-dev`) was added to support
this transitional use case, so we employ it here.

Exceptions:
Of note, the ci-kubernetes-e2e-gce-master-new-gci-kubectl-skew-serial is
not included in this PR.

This job does two extractions:
- `--extract=ci/k8s-stable1`
- `--extract=ci/latest-fast`

As the generic version markers (like `k8s-stable1`) have not been
migrated to K8s Infra, we cannot take advantage of this flag.

We'll plan to fix this job in a follow-up.

(May mitigate https://github.com/kubernetes/test-infra/issues/19838.)

This is part of migrating release-blocking jobs to K8s Infra (ref: https://github.com/kubernetes/test-infra/issues/19484, https://github.com/kubernetes/test-infra/issues/18549).

---

Previous thought process (https://github.com/kubernetes/test-infra/commit/2b282b3c0a01cb02609ed27ac8f73c5d7e3567a8) is explained here:

~This reverts commit f1fd4142d87e877832be19d89037214d917b76aa. (ref: https://github.com/kubernetes/test-infra/pull/19660).~

~tl;dr of what's happening is that any existing job that extracts the `latest-fast` version marker is extracting the marker from `gs://kubernetes-release-dev/ci/fast` (Google Infra) instead of `gs://k8s-release-dev/ci/fast` (K8s Infra), which means they are using stale builds.~

~We need to make an accompanying change to the extract logic, which will take some time to roll out new images for.
This is the quickest course of action in the meantime.~

/assign @cpanato @saschagrunert @hasheddan 
/priority critical-urgent
cc: @kubernetes/sig-scalability @kubernetes/release-engineering

### CODE REVIEW

**Comment by justaugustus** [comment]
@kubernetes/ci-signal -- please keep an eye on these jobs.

**Comment by justaugustus** [comment]
Actually... @spiffxp may have already provided a mechanism for fixing this with the 

> Now that #19634 has confirmed the new bucket works, remaining steps are:
> 
>     * wait for #19631 to merge or bump kubekins in jobs to be migrated (see #19632)
> 
>     * any jobs that have `--extract=ci/latest-fast` should have `--extract-ci-bucket=k8s-release-dev` added (see #19634)
> 
>     * conform migrated jobs work
> 
>     * update canary to match old job name and annotations, delete old job

Let me rework this PR.
/hold

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

File: config/jobs/kubernetes/sig-cloud-provider/gcp/gce-conformance.yaml
Comment (hasheddan) [inline config/jobs/kubernetes/sig-cloud-provider/gcp/gce-conformance.yaml:26]: @justaugustus do we always want to get version markers from `k8s-release-dev` now, or are some jobs (not necessarily owned by sig-release) still requiring usage of `kubenertes-release-dev`?

Comment (justaugustus) [reply config/jobs/kubernetes/sig-cloud-provider/gcp/gce-conformance.yaml:26]: @hasheddan -- explained in the updated description, but pasting to fire the email notification:

> As we continue to migrate release-blocking jobs to a dedicated K8s Infra
cluster, jobs that use the latest-fast marker need to extract builds
from gs://k8s-release-dev, which is the K8s Infra equivalent of
gs://kubernetes-release-dev.
>
> A new flag (--extract-ci-bucket=k8s-release-dev) was added to support
this transitional use case, so we employ it here.

Comment (hasheddan) [reply config/jobs/kubernetes/sig-cloud-provider/gcp/gce-conformance.yaml:26]: @justaugustus gotcha 👍 was just wondering if we were ready to modify the extract logic to use this by default

**Review by xmudrii — APPROVED** [review]
/lgtm

**Comment by justaugustus** [comment]
Need a `config/` approver now...
/assign @dims

**Review by hasheddan — APPROVED** [review]
/lgtm

File: config/jobs/kubernetes/sig-cloud-provider/gcp/gce-conformance.yaml
Comment (justaugustus) [reply config/jobs/kubernetes/sig-cloud-provider/gcp/gce-conformance.yaml:26]: @hasheddan -- Not yet. The primary (non-fast) build jobs still need to be migrated. That's being tracked in https://github.com/kubernetes/test-infra/issues/19483.

**Comment by wojtek-t** [comment]
/hold

@justaugustus - are we sure that this will fix all jobs?

**Comment by wojtek-t** [comment]
/hold cancel

copying from slack discussion:

> so I guess it only impacts those that are using "--extract=ci/latest-fast" (as opposed to "--extract=ci/latest" which is the majority of jobs are still using), right?
> exactly :slightly_smiling_face:

**Comment by justaugustus** [comment]
> @justaugustus - are we sure that this will fix all jobs?

Fairly certain!
`ci-kubernetes-e2e-gci-gce` is already using this flag and you can see from its [most recent successful run](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce/1323988815383105536/) that it's using a build version a few commits ahead on `v1.20.0-beta.1` (which was released yesterday).

```console
W1104 14:02:14.106] 2020/11/04 14:02:14 extract_k8s.go:295: U=https://storage.googleapis.com/k8s-release-dev/ci/fast R=v1.20.0-beta.1.49+71fea80155e5c1 get-kube.sh
W1104 14:02:14.107] 2020/11/04 14:02:14 process.go:153: Running: ./get-kube.sh
I1104 14:02:14.207] Downloading kubernetes release v1.20.0-beta.1.49+71fea80155e5c1
I1104 14:02:14.207]   from https://storage.googleapis.com/k8s-release-dev/ci/fast/v1.20.0-beta.1.49+71fea80155e5c1/kubernetes.tar.gz
I1104 14:02:14.208]   to /workspace/kubernetes.tar.gz
W1104 14:02:15.408] Copying gs://k8s-release-dev/ci/fast/v1.20.0-beta.1.49+71fea80155e5c1/kubernetes.tar.gz...
W1104 14:02:15.559] / [0 files][    0.0 B/498.3 KiB]                                                
/ [1 files][498.3 KiB/498.3 KiB]                                                
W1104 14:02:15.560] Operation completed over 1 objects/498.3 KiB.                                    
I1104 14:02:15.821] Unpacking kubernetes release v1.20.0-beta.1.49+71fea80155e5c1
I1104 14:02:15.889] Kubernetes release: v1.20.0-beta.1.49+71fea80155e5c1 
```

**Pull request merged**
